app/controllers/histories_controller.rb

```diff
@@ -8,9 +8,45 @@ class HistoriesController < ApplicationController
   def create
     redirect_to root_url unless current_user.can_create_subject_history?
 
-    History.create!(patient_id: params.permit(:patient_id)[:patient_id],
+    history = History.new(patient_id: params.permit(:patient_id)[:patient_id],
```

Comment:
need to check if patient is viewable by user by adding
```
patient = current_user.viewable_patients.find_by(id: params.permit(:patient_id)[:patient_id])
```

Reply:
https://github.com/SaraAlert/SaraAlert/pull/793/commits/c8f1f62444e06ae8640e77e29be166045165d6dd#diff-c8c8197061d8db322cb11806e49c59c00413cfb234997e88adfe4e939477b69aR11